src/AutoRest.CSharp/Common/Output/Models/ClientMethodBodyLines.cs

```diff
@@ -0,0 +1,216 @@
+            {
+                var requestMethodCall = new InstanceMethodCallExpression(restClient, methodName, arguments.ToList(), false);
+                localFunctionName = new CodeWriterDeclaration("FirstPageRequest");
+                return new OneLineLocalFunction(localFunctionName, new[]{KnownParameters.PageSizeHint}, typeof(HttpMessage), requestMethodCall);
+            }
+
+            public static MethodBodyLine NextPageRequest(ValueExpression? restClient, string methodName, IEnumerable<ValueExpression> arguments, out CodeWriterDeclaration localFunctionName)
+            {
+                var requestMethodCall = new InstanceMethodCallExpression(restClient, methodName, arguments.ToList(), false);
+                localFunctionName = new CodeWriterDeclaration("NextPageRequest");
+                return new OneLineLocalFunction(localFunctionName, new[]{KnownParameters.PageSizeHint, KnownParameters.NextLink}, typeof(HttpMessage), requestMethodCall);
+            }
+
+            public static MethodBodyLine Message(ValueExpression value, out CodeWriterDeclaration message)
+            {
+                message = new CodeWriterDeclaration("message");
+                return new UsingDeclareVariableLine(typeof(HttpMessage), message, value);
+            }
+
+            public static MethodBodyLine Response(CSharpType responseType, ValueExpression value, out CodeWriterDeclaration response)
+            {
+                response = new CodeWriterDeclaration("response");
+                return new DeclareVariableLine(responseType, response, value);
+            }
+
+            public static MethodBodyLine RequestContext(ValueExpression value, out CodeWriterDeclaration requestContext)
+            {
+                requestContext = new CodeWriterDeclaration(KnownParameters.RequestContext.Name);
+                return new DeclareVariableLine(KnownParameters.RequestContext.Type, requestContext, value);
+            }
+
+            public static MethodBodyLine NewModelInstance(ModelTypeProvider model, IEnumerable<Parameter> arguments, out CodeWriterDeclaration variable)
+                => NewModelInstance(model, arguments.Select(p => new ParameterReference(p)).ToList(), out variable);
+
+            public static MethodBodyLine NewModelInstance(ModelTypeProvider model, IReadOnlyList<ValueExpression> arguments, out CodeWriterDeclaration variable)
+            {
+                variable = new CodeWriterDeclaration(model.Type.Name.ToVariableName());
+                var newInstance = new NewInstanceExpression(model.Type, arguments);
+                return new DeclareVariableLine(model.Type, variable, newInstance);
+            }
+        }
+    }
+
+    internal record Method(MethodSignature Signature, MethodBody Body);
+    internal record MethodBody(IReadOnlyList<MethodBodyBlock> Blocks);
+
+    internal record MethodBodyBlock
```

Comment:
I typically see a code block as the set of lines that go inside a for, if, else etc.  If this is the case would we not represent this as an having an opening line as well as a set of lines that go inside?

usage would be something like
```c#
public static WriteMethodBodyBlock(this CodeWriter writer, MethodBodyBlock methodBodyBlock)
{
    writer.Scope(methodBodyBlock.OpeningLine)
    {
        foreach(var line in methodBodyBLock.Lines)
        {
            writer.Line(line);
        }
    }
}
```

If instead this isn't intended to represent a block with this definition can we change it to `MethodBodySnippet` to simply represent some subset of the entire method body?

Reply:
Unlike with `ValueExpression` where we have basically any type of supported C# expression, we only need a very small subset of code blocks:
 - foreach in deserialization
 - try/catch/finally for DiagnosticsScope
 - switch for status code handling

So we don't really need fully generalized method body block support. Maybe call them `MethodBodyLinesSnippets`?

Reply:
I don't think `Lines` adds anything we can probably shrink to just `MethodBodySnippet`